book/translation.rst

```diff
@@ -428,6 +428,14 @@ via the ``request`` object::
 
         $request->setLocale('en_US');
     }
+    
+.. note::
+
+    Setting the locale using the ``$request->setLocale()`` method won't affect 
+    rendering in the same action. The translator locale is set during the
+    ``kernel.request`` event. Either set the locale before the listener is called
+    (e.g. in a custom listener) or use the ``setLocale()`` method of the 
+    ``translator`` service.
```

Comment:
It does feel hackish, but I can't really see why it shouldn't be done this way. It works perfectly and I could not find any other way to change the local of the translator in the controller. 

I believe it feels hackish, because the translator service is normally not accessed, so this means another depency. This shouldn't cause any problems though.

> * Remove the setLocale in the code block above
> * Add another code block showing an event listener that is calling setLocale, and just mention quickly that this would be an event listener on kernel.request, with a link to the event listener sections.

This is a great idea! I'll sketch a new version as soon as I can and push it, see if you like it.